src/components/NavBar/ShoppingBag.tsx

```diff
@@ -2,14 +2,33 @@ import { NavIcon } from 'components/NavBar/NavIcon'
 import * as styles from 'components/NavBar/ShoppingBag.css'
 import { Box } from 'nft/components/Box'
 import { BagIcon, HundredsOverflowIcon } from 'nft/components/icons'
-import { useBag } from 'nft/hooks'
-import { useEffect, useState } from 'react'
+import { useBag, useSellAsset } from 'nft/hooks'
+import { useCallback, useEffect, useState } from 'react'
+import shallow from 'zustand/shallow'
 
 export const ShoppingBag = () => {
   const itemsInBag = useBag((state) => state.itemsInBag)
   const [bagQuantity, setBagQuantity] = useState(0)
 
-  const toggleBag = useBag((s) => s.toggleBag)
+  const { bagExpanded, setBagExpanded } = useBag(
```

Comment:
this part i'm not actually following

Reply:
zustand state read/write util `useBag` updated to provide this component with both current state and a write method (so can retain toggling behavior, and add conditional behavior based on current state.